pkgs/development/python-modules/versioningit/default.nix

```
@@ -0,0 +1,63 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchPypi
+, importlib-metadata
+, packaging
+, setuptools
+, tomli
+, pytestCheckHook
+, build
+, pydantic
+, pytest-cov
+, pytest-mock
+, git
+, mercurial
+}:
+
+buildPythonPackage rec {
+  pname = "versioningit";
+  version = "2.0.0";
+  disabled = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-wWkTrzW76WeXuqOgkQ/GCxwF0lrFkd72Ij+oBvmcvlg=";
+  };
+
+  format = "pyproject";
```

```suggestion
  disabled = pythonOlder "3.8";
  format = "pyproject";

  src = fetchPypi {
    inherit pname version;
    sha256 = "sha256-wWkTrzW76WeXuqOgkQ/GCxwF0lrFkd72Ij+oBvmcvlg=";
  };

```
most common to punch all of the "high level" configuration toward the top